packages/zipkin/src/internalRepresentations.js

```diff
@@ -29,6 +30,17 @@ Endpoint.prototype.toJSON = function toJSON() {
   };
 };
 
+Endpoint.createEndpoint = function createEndpoint({serviceName, host, port}, callback) {
+  getAddressByName(host, addr => {
```

Comment:
what I mean is that the host (or actually what's derived from it which is either an ipv4 or ipv6 address) is lower value data. The service name is the most important thing, so it would be nice to contain the fanciness to the host. In other words, a failure to parse what we are calling "host" shouldn't make endpoint fail to be set.